sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -341,69 +343,160 @@ void initializeHttpRequest(BlockStmt body, HttpRequestContext method) {
     }
 
     private void appendQueryParams(BlockStmt body, HttpRequestContext method) {
-        // Iterate through the query parameters and append them to the url string if they are not null
-        if (!method.getQueryParams().isEmpty()) {
-            // Declare newUrl once
-            Statement newUrlDeclaration = StaticJavaParser.parseStatement("String newUrl;");
-            newUrlDeclaration.setComment(new LineComment("\n Append non-null query parameters"));
-            body.addStatement(newUrlDeclaration);
-            body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
-            body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
-
-            method.getQueryParams().entrySet().forEach(entry -> {
-                String key = entry.getKey();
-                HttpRequestContext.QueryParameter value = entry.getValue();
-                boolean isValueTypeString = method.getParameters()
-                    .stream()
-                    .anyMatch(parameter -> parameter.getName().equals(value.getValue())
-                        && "String".equals(parameter.getShortTypeName()));
-                if (value.shouldEncode()) {
-                    if (isValueTypeString) {
-                        String encodedKey = "UriEscapers.QUERY_ESCAPER.escape(\"" + key + "\")";
-                        String encodedValue = "UriEscapers.QUERY_ESCAPER.escape(" + value.getValue() + ")";
+        if (method.getQueryParams().isEmpty()) {
+            return;
+        }
+
+        // Add imports and declare variables
+        body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
+        body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
+
+        // Iterate through query parameters
+        method.getQueryParams().forEach((key, value) -> {
+            if (CoreUtils.isNullOrEmpty(key)) {
+                // Skip null or empty keys
+                return;
+            }
+
+            String valueExpression = value.getValue();
+            boolean isValueTypeString = method.getParameters()
+                .stream()
+                .anyMatch(parameter -> parameter.getName().equals(valueExpression)
+                    && "String".equals(parameter.getShortTypeName()));
+            boolean isValueTypeListOfString = method.getParameters()
+                .stream()
+                .anyMatch(parameter -> parameter.getName().equals(valueExpression)
+                    && "List<String>".equals(parameter.getShortTypeName()));
+
+            if (!CoreUtils.isNullOrEmpty(valueExpression)) {
+                String encodedKey = "UriEscapers.QUERY_ESCAPER.escape(\"" + key + "\")";
```

Comment:
Shouldn't key encoding be based on value encoding?

Though really, we should have a way to indicate both separately.

Reply:
Talked offline, decided that it might be an overkill to have to indicate encoding for both key and value separately. And also, that it wouldn't hurt to encode both key and value when encoding = true.